src/gflownet/train.py

```diff
@@ -100,21 +105,23 @@ def __init__(self, hps: Dict[str, Any], device: torch.device):
         # `sampling_model` is used by the data workers to sample new objects from the model. Can be
         # the same as `model`.
         self.sampling_model: nn.Module
-        self.replay_buffer: ReplayBuffer
+        self.replay_buffer: Optional[ReplayBuffer]
         self.mb_size: int
         self.env: GraphBuildingEnv
         self.ctx: GraphBuildingEnvContext
         self.task: GFNTask
         self.algo: GFNAlgorithm
 
-        # Override default hyperparameters with the constructor arguments
-        self.hps = {**self.default_hps(), **hps}
+        # There are three sources of config values
+        #   - The default values specified in individual config classes
+        #   - The default values specified in the `default_hps` method, typically what is defined by a task
+        #   - The values passed in the constructor, typically what is called by the used
+        # The final config is obtained by merging the three sources
+        self.cfg = OmegaConf.structured(Config())
+        self.set_default_hps(self.cfg)
+        OmegaConf.merge(self.cfg, hps)
```

Comment:
Given the complexity of the config, I would suggest to consider adding a linter, just to make sure that user provided configs are compliant to the expected structure (avoid missing key error, overwriting of default, etc)